derrida/interventions/management/commands/insertion_data.py

```diff
@@ -0,0 +1,145 @@
+import csv
+import json
+import os.path
+import re
+from itertools import groupby
+
+from django.db.models import ObjectDoesNotExist
+from djiffy.models import Canvas
+
+from derrida.books.management.commands import reference_data
+from derrida.interventions.models import Intervention
+
+
+# regex to extract insertion label common to all images
+# for a single insertion from the canvas label
+# grouping for full label, page range/label in book, insertion page label
+# FIXME: negative lookahead for with in page number is not working properly
+RE_INSERTION_LABEL = re.compile(r'(?P<label>(?P<page>.*(?! with))( with)?Insertions? [A-Z])(?P<insertion_page>.*$)')
+
+
+class Command(reference_data.Command):
+    '''Export intervention insertion data from the database as CSV and JSON'''
+    help = __doc__
+
+    # NOTE: extending reference_data manage command to inherit
+    # flatten_data method; there is more overlap and these scripts
+    # could probably be generalized further for re-use
+
+    #: fields for CSV output
+    csv_fields = [
+        # match annotation fields where possible
+        'id', 'intervention_type',
+        'book_id', 'book_title', 'book_type', 'page',
+        #, 'tags', 'text content',
+        #'text language', 'text language code', 'text translation',
+        #'quote content', 'quote language', 'quote language code', 'annotator'
+        'num_images', 'image_labels', 'image_iiif'
+    ]
+
+    #: base filename, for CSV and JSON output
+    base_filename = 'insertions'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-d', '--directory',
+            help='Specify the directory where files should be generated')
+
+    def handle(self, *args, **kwargs):
+
+        if kwargs['directory']:
+            self.base_filename = os.path.join(kwargs['directory'], self.base_filename)
+
+        # aggregate intervention data to be exported for use in generating
+        # CSV and JSON output
+
+        # canvas label indicates if a canvas is part of an insertion
+        insertion_canvases = Canvas.objects.filter(label__contains='Insertion')
+
+        insertion_images = defaultdict(list)
+        for canvas in insertion_canvases:
+            insertion_match = RE_INSERTION_LABEL.match(canvas.label)
+            if not insertion_match:
+                print('ERROR: regex fails on %s' % canvas.label)
+                continue
+
+            # construct group label from manifest label and insertion title within canvas label
+            # so we get unique image groups for each logical insertion
+            group_label = '%s. %s' % (canvas.manifest.label.strip('.'), insertion_match.group('label'))
+            insertion_images[group_label].append(canvas)
+
+        data = [self.insertion_data(label, canvas_group)
+                for label, canvas_group in insertion_images.items()]
+
+        # filter out any null values for canvases not linked to work instance
+        data = [d for d in data if d]
```

Comment:
I don't think this does what your comment says. Won't `d` always be cast as true? (a non-empty Ordered dict)